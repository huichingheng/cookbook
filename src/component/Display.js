import React from "react";
import "../style/Display.css";
import Modal from "./Modal";
import percentToRating from "../utils/percentToRating"

const Display = props => {
  const { selectedRecipe } = props;
  const RATING_STEPS = 5

  if (selectedRecipe) {
    return (
      <div className="display">
        <div className="title">{selectedRecipe.title}</div>
        <div className="rating">
         {percentToRating(selectedRecipe.social_rank, RATING_STEPS)}
        </div>
        <img className="display-image" src={selectedRecipe.image_url} alt="" />
        <Modal id="openModal" selectedRecipe={selectedRecipe} />
        <div className="recipe-source">
          Source: {selectedRecipe.publisher}, {selectedRecipe.publisher_url}
        </div>
      </div>
    );
  }
  return <div className="display" />;
};

export default Display;
