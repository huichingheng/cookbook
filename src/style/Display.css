.display {
  grid-column: 2/3;
  border: 1px solid black;
  box-sizing: border-box;
}

.display .content {
  font-size: 0.75em;
}

.display .content-empty {
  font-size: 0.75em;
  box-sizing: border-box;
}

.display .title {
  font-family: "Gill Sans MT", Calibri, "Trebuchet MS", sans-serif;
  font-weight: bold;
  font-size: calc(10px + 1vw);
  margin: 20px;
  box-sizing: border-box;
}

.recipe-source {
  font-size: 0.75em;
  margin: 20px;
  box-sizing: border-box;
}

.display-image {
  background: 50% 50% no-repeat;
  object-fit: contain;
  margin: 5px auto;
  width: 250px;
  height: auto;
  font-size: 0.75em;
  box-sizing: border-box;
}

@media all and (max-width: 850px) {
  .display {
    grid-column: 1/2;
    height: 60vh;
  }

  .display-image {
    width: 200px;
  }

  .display .title {
    margin: 10px;
  }

  .recipe-source {
    margin: 10px;
    font-size: 0.6em;
  }
}
