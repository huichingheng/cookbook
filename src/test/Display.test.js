import React from "react";
import { shallowRender } from "./helper/testHelper";
import { shallow } from "enzyme";
import Display from "../component/Display";
import testData from "./helper/testData"


const wrapper = shallow(
  <Display selectedRecipe={testData} />
);

describe("<Display />", () => {
  describe("should have", () => {
    it("title", () => {
      expect(wrapper.find(".title")).toHaveLength(1);
    });
    it("image", () => {
      expect(wrapper.find(".display-image")).toHaveLength(1);
    });
    it("recipe link", () => {
      expect(wrapper.find(".recipe-link")).toHaveLength(1);
    });
    it("link text", () => {
      expect(wrapper.find(".popularity")).toHaveLength(1);
    });
    it("link url", () => {
      expect(wrapper.find(".link-url")).toHaveLength(1);
    });
  });

  describe("Snapshot", () => {
    const result = shallowRender(Display);
    it("should match", () => {
      expect(result.props).toMatchSnapshot();
    });
  });
});
