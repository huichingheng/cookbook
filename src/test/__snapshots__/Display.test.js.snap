// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> Snapshot should match 1`] = `
Object {
  "children": Array [
    <br />,
    <div
      className="content-empty"
>
      <div>
            Search for recipes using the search bar.
      </div>
      <p />
      Seperate ingredients by commas.
      <div>
            Example: chicken, onion, carrot, pepper
      </div>
</div>,
  ],
  "className": "display",
}
`;
